presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -233,8 +233,15 @@ void PrestoServer::run() {
   registerMemoryArbitrators();
   registerShuffleInterfaceFactories();
   registerCustomOperators();
-  protocol::registerHiveConnectors();
-  protocol::registerTpchConnector();
+
+  registerPrestoToVeloxConnector(
+      std::make_unique<HivePrestoToVeloxConnector>("hive"));
```

Comment:
I would expect a single instance of `HivePrestoToVeloxConnector` to be used for both 
`hive` and `hive-hadoop2`. The unique_ptr choice seems odd to me.
The ConnectorProtocol translates the protocol structures from Presto to Velox. There is no state.
I would expect a singleton ConnectorProtocol instance for each Velox connector and not for each Presto Connector.

Reply:
Here's how I'm thinking about it.

PrestoToVeloxConnector is a virtual interface and must be interacted with via a pointer. 

When making this choice I considered:

- Raw pointer

The method would expect the lifecycle of a `PrestoToVeloxConnector` managed elsewhere and could be error prone.

It is probably fine if implementations are expected to be statically allocated. However it is hard to say whether it will always be the case for every possible connectors. I can see a case when an implementation of a `PrestoToVeloxConnector` is configured based on runtime parameters / configuration.

- Shared pointer

It is quite unusual to have two exactly same instances of `PrestoToVeloxConnector`. The `hive-hadoop2` connector exists for purely legacy reasons. Wasn't sure if it's worth expressing in the interface.

With these considerations I decided to simply go with a unique pointer.

Reply:
>  It is probably fine if implementations are expected to be statically allocated. However it is hard to say whether it will always be the case for every possible connectors. I can see a case when an implementation of a PrestoToVeloxConnector is configured based on runtime parameters / configuration.

This is not possible in the current setup since the presto_protocol structures are static. Dynamic connector support would need a new design of the protocol as well.

Reply:
> This is not possible in the current setup since the presto_protocol structures are static. Dynamic connector support would need a new design of the protocol as well.

Could you please elaborate?

I can conceivably see someone adding a connector taking a configuration property. For example the same very line could've been like:

```
...
registerPrestoToVeloxConnector(
      std::make_unique<HivePrestoToVeloxConnector>("hive", systemConfig->ignoreUnsupportedPrestoFilters()));
registerPrestoToVeloxConnector(
      std::make_unique<HivePrestoToVeloxConnector>("hive-hadoop2", true));
...
```

Basically an instance of a `HivePrestoToVeloxConnector` is created with a certain configuration in mind.

The current connectors don't take any configuration options, but some other connectors potentially could.

Reply:
I misunderstood your initial statement.
If you meant dynamic loading of catalog properties,
Ddon't we already load config from the `.properties` file and also session level properties?
When would we need this?

Reply:
Will this change the protocol translation?

Reply:
Currently it is not needed for existing connectors. It may be needed in the future. I can see a use for this pattern for an internal connector I'm trying to build at Meta.